src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs

```diff
@@ -554,5 +560,71 @@ public void EditIpo_ShouldAddEditIpoEvent()
         [TestMethod]
         public void Constructor_ShouldAddIpoCreatedEvent() 
             => Assert.IsInstanceOfType(_dutWithMcPkgScope.DomainEvents.First(), typeof(IpoCreatedEvent));
+
+        [TestMethod]
+        public void Cancel_SetsStatusToCanceled()
+        {
+            TimeService.SetProvider(new ManualTimeProvider(new DateTime(2021, 1, 1, 12, 0, 0, DateTimeKind.Utc)));
+            var creator = new Person(new Guid("12345678-1234-1234-1234-123456789123"), "Test", "Person", "tp", "tp@pcs.pcs");
+
+            var dut = new Invitation(
+                TestPlant,
+                ProjectName,
+                Title,
+                Description,
+                DisciplineType.MDP,
+                new DateTime(2020, 8, 1, 12, 0, 0, DateTimeKind.Utc),
+                new DateTime(2020, 8, 1, 13, 0, 0, DateTimeKind.Utc),
+                null);
+
+            dut.SetCreated(creator);
+
+            dut.Cancel(creator);
+            Assert.AreEqual(dut.Status, IpoStatus.Canceled);
+        }
+
+        [TestMethod]
+        public void Cancel_CallerIsNotCreator_ThrowsException()
+        {
+            TimeService.SetProvider(new ManualTimeProvider(new DateTime(2021, 1, 1, 12, 0, 0, DateTimeKind.Utc)));
```

Comment:
Do we need these timeservices?

Reply:
Yes, because we need to set the CreatedById property, and the SetCreated function also sets the CreatedAt properties.